addons/account_edi_ubl/models/account_edi_format_en_16931.py

```diff
@@ -0,0 +1,243 @@
+            'customization_id': 'urn:cen.eu:en16931:2017#compliant#urn:fdc:peppol.eu:2017:poacc:billing:3.0',
+            'profile_id': 'urn:fdc:peppol.eu:2017:poacc:billing:01:1.0',
+            'due_date': fields.Date.to_string(invoice.invoice_date_due),
+            'invoice_date': fields.Date.to_string(invoice.invoice_date),
+
+            'convert_monetary': convert_monetary,
+            'partner_template': 'account_edi_ubl.export_en_16931_invoice_partner',
+        })
+
+        # Compute tax and price values based on compute_all
+        # In BIS3 only vat are considered as taxes, other taxes must be applied already.
+        values['lines'] = {}
+        values['price_subtotal'] = 0
+        values['tax_total'] = 0
+        values.setdefault('breakdown', {})
+        for line in invoice.invoice_line_ids.filtered(lambda line: not line.display_type):
+            price_unit_wo_discount = line.price_unit * (1 - (line.discount / 100.0))
+            line_taxes = line.tax_ids._origin.compute_all(
+                price_unit_wo_discount,
+                quantity=line.quantity,
+                product=line.product_id,
+                partner=invoice.partner_id,
+                currency=line.currency_id,
+                is_refund=invoice.move_type in ('out_refund', 'in_refund'))
+            line_vals = {'base': line_taxes['total_excluded'], 'category': 'Z', 'percent': 0}
+            values['price_subtotal'] += line_taxes['total_excluded']
+            for tax in line_taxes['taxes']:
+                repartition_line = self.env['account.tax.repartition.line'].browse(tax['tax_repartition_line_id'])
+                if repartition_line.use_in_tax_closing:
+                    tax_percent = self.env['account.tax'].browse(tax['id']).amount
+                    tax_category = 'S' if tax_percent else 'Z'
+                    if line_vals['category'] != 'Z' and line_vals['percent'] != tax_percent:
+                        raise UserError('Multiple vat percentage not supported on the same invoice line')
+                    else:
+                        line_vals['percent'] = tax_percent
+                        line_vals['category'] = tax_category
+                        values['breakdown'].setdefault((tax_category, tax_percent), {'base': 0, 'amount': 0})
+                        values['breakdown'][(tax_category, tax_percent)]['base'] += tax['base']
+                        values['breakdown'][(tax_category, tax_percent)]['amount'] += tax['amount']
+                        values['tax_total'] += tax['amount']
+                else:
+                    line_vals['base'] += tax['amount']
+                    values['price_subtotal'] += tax['amount']
+            values['lines'][line] = line_vals
+
+        values['seller_currency'] = invoice.company_id.currency_id,
+        values['tax_total_seller_currency'] = invoice.currency_id._convert(values['tax_total'], invoice.company_id.currency_id, invoice.company_id, invoice.date),
```

Comment:
This is confusing. If you need the tax details in company's currency, you don't need a conversion since you could use this currency when calling `compute_all`. I'm worry about rounding issue there: can't you provide taxes amounts in invoice's currency?